newcastle_racing_ai/newcastle_racing_ai/controller.py

```diff
@@ -5,22 +5,52 @@
 from ackermann_msgs.msg import AckermannDriveStamped, AckermannDrive
 from .parameters import PARAMETERS
 
+
 class Controller(Node):
+    """A value that means `do nothing`"""
+
+    __null_ads = AckermannDriveStamped(
+        drive=AckermannDrive(
+            steering_angle=0.0,
+            steering_angle_velocity=0.0,
+            speed=0.0,
+            acceleration=0.0,
+            jerk=0.0,
+        )
+    )
 
     def __init__(self):
-        super().__init__('Controller')
+        super().__init__("Controller")
         self.declare_parameters(namespace="", parameters=PARAMETERS)
-        self._subscription = self.create_subscription(PoseArray, self.get_parameter("path_topic").value, self._on_path, 10)
-        self._publisher = self.create_publisher(AckermannDriveStamped, self.get_parameter("cmd_topic").value, 10)
+        self._subscription = self.create_subscription(
+            PoseArray, self.get_parameter("path_topic").value, self._on_path, 10
+        )
+        self._publisher = self.create_publisher(
+            AckermannDriveStamped, self.get_parameter("cmd_topic").value, 10
+        )
         # self.timer = self.create_timer(timer_period, self._timer_callback)
 
+        self._steering_angle = 0.0
+        self._steering_angle_velocity = 0.0
+        self._speed = 0.0
+        self._acceleration = 0.0
+        self._jerk = 0.0
+
+        self.ads = self.__null_ads
```

Comment:
I think you can just keep the instance of `AckermannDriveStamped` in `self.ads`. You can read its fields and/or update them as needed. I don't think you also need to store (and update every time) all these

```py
        self._steering_angle = 0.0
        self._steering_angle_velocity = 0.0
        self._speed = 0.0
        self._acceleration = 0.0
        self._jerk = 0.0
```